repos/spack_repo/builtin/packages/pandoc/package.py

```diff
@@ -50,22 +57,43 @@ class Pandoc(Package):
             url="https://github.com/jgm/pandoc/releases/download/2.7.3/pandoc-2.7.3-linux.tar.gz",
         )
 
+    elif platform.system() == "Darwin" and platform.machine() == "aarch64":
+        url = "https://github.com/jgm/pandoc/releases/download/2.14.0.3/pandoc-3.7.0.2-arm64-macOS.zip"
+
+        version(
+            "3.7.0.2", sha256="66a579bd8aae83de0bbeba43900953b075a6a3caaa7d1bfc19173e8f95d2ea17"
+        )
+
     elif platform.system() == "Darwin":
-        url = "https://github.com/jgm/pandoc/releases/download/2.14.0.3/pandoc-2.14.0.3-macOS.zip"
+        url = "https://github.com/jgm/pandoc/releases/download/2.14.0.3/pandoc-3.7.0.2-x86_64-macOS.zip"
```

Comment:
The version has to be replaced in both locations for the substitution to work as expected:

```suggestion
        url = "https://github.com/jgm/pandoc/releases/download/3.7.0.2/pandoc-3.7.0.2-x86_64-macOS.zip"
```